### PR TITLE
Fix: Sea Creature catch messages not using proper articles

### PR DESCRIPTION
**Pull request by lunaynx**
## What
`StringUtils.optionalAn` was only returning `"an"` if the string starts with a *lowercase* vowel; otherwise it would return `"a"`. The only practical user-facing effect of this today is that Sea Creature catch messages are wrong: "I caught a Oasis Sheep!" rather than "I caught an Oasis Sheep!".

## Changelog Fixes
+ Fixed Sea Creature catch messages incorrectly saying "a" instead of "an" (e.g. "a Oasis Sheep"). - Luna

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt [hunk 1]
```diff
@@ -527,7 +527,7 @@ object StringUtils {
 
     fun optionalAn(string: String): String {
         if (string.isEmpty()) return ""
-        return if (string[0] in "aeiou") "an" else "a"
+        return if (string[0].lowercase() in "aeiou") "an" else "a"
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt]: `Char.isVowel` exists already, couple lines above

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt]: `EventHandler.post` uses this too, so error messages change as well.

File: src/test/java/at/hannibal2/skyhanni/test/StringUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+class StringUtilsTest {
+    @Test
+    fun `optionalAn returns the correct article`() {
+        assertEquals("an", StringUtils.optionalAn("apple"))
+        assertEquals("an", StringUtils.optionalAn("Apple"))
+        assertEquals("a", StringUtils.optionalAn("pear"))
+        assertEquals("a", StringUtils.optionalAn("Pear"))
+        assertEquals("a", StringUtils.optionalAn("yeti"))
+        assertEquals("a", StringUtils.optionalAn("Yeti"))
```
Comment (hannibal002) [inline src/test/java/at/hannibal2/skyhanni/test/StringUtilsTest.kt:15]: Also test the empty string.

**Review by hannibal002 — CHANGES_REQUESTED** [review]

**Review by hannibal002 — APPROVED** [review]
